### ISSUE TITLE
Not drawing correctly on MacOS catalina

### ISSUE THREAD

**Issue by freesig** [nannou-org/nannou#423]
It looks like the draw order is wrong when simple_draw is run.
Intel Irish Plus Graphics 650

**Comment by teh-f00l** [comment]
<img width="815" alt="image" src="https://user-images.githubusercontent.com/36428156/68531094-13cfce00-030f-11ea-8bf6-75f86a328b40.png">

**Comment by teh-f00l** [comment]
```
$ ./vulkaninfo 
==========
VULKANINFO
==========

Vulkan Instance Version: 1.1.126


Instance Extensions: count = 9
====================
        VK_EXT_debug_report                    : extension revision 9
        VK_EXT_debug_utils                     : extension revision 1
        VK_EXT_metal_surface                   : extension revision 1
        VK_EXT_swapchain_colorspace            : extension revision 4
        VK_KHR_device_group_creation           : extension revision 1
        VK_KHR_get_physical_device_properties2 : extension revision 2
        VK_KHR_get_surface_capabilities2       : extension revision 1
        VK_KHR_surface                         : extension revision 25
        VK_MVK_macos_surface                   : extension revision 2

Layers: count = 3
=======
VK_LAYER_KHRONOS_validation (Khronos Validation Layer) Vulkan version 1.1.126, layer version 1:
        Layer Extensions: count = 3
                VK_EXT_debug_report        : extension revision 9
                VK_EXT_debug_utils         : extension revision 1
                VK_EXT_validation_features : extension revision 2
        Devices: count = 1
                GPU id  : 0 (Intel(R) Iris(TM) Plus Graphics 650)
                Layer-Device Extensions: count = 2
                        VK_EXT_debug_marker     : extension revision 4
                        VK_EXT_validation_cache : extension revision 1

VK_LAYER_LUNARG_api_dump (LunarG API dump layer) Vulkan version 1.1.126, layer version 2:
        Layer Extensions: count = 0
        Devices: count = 1
                GPU id  : 0 (Intel(R) Iris(TM) Plus Graphics 650)
                Layer-Device Extensions: count = 0

VK_LAYER_LUNARG_standard_validation (LunarG Standard Validation) Vulkan version 1.1.126, layer version 1:
        Layer Extensions: count = 3
                VK_EXT_debug_report        : extension revision 9
                VK_EXT_debug_utils         : extension revision 1
                VK_EXT_validation_features : extension revision 2
        Devices: count = 1
                GPU id  : 0 (Intel(R) Iris(TM) Plus Graphics 650)
                Layer-Device Extensions: count = 2
                        VK_EXT_debug_marker     : extension revision 4
                        VK_EXT_validation_cache : extension revision 1

Presentable Surfaces:
=====================
GPU id : 0 (Intel(R) Iris(TM) Plus Graphics 650):
        Surface types: count = 1
                VK_MVK_macos_surface
        Formats: count = 48
                SurfaceFormat[0]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_SRGB_NONLINEAR_KHR
                SurfaceFormat[1]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_SRGB_NONLINEAR_KHR
                SurfaceFormat[2]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_SRGB_NONLINEAR_KHR
                SurfaceFormat[3]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_SRGB_NONLINEAR_KHR
                SurfaceFormat[4]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_DISPLAY_P3_NONLINEAR_EXT
                SurfaceFormat[5]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_DISPLAY_P3_NONLINEAR_EXT
                SurfaceFormat[6]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_DISPLAY_P3_NONLINEAR_EXT
                SurfaceFormat[7]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_DISPLAY_P3_NONLINEAR_EXT
                SurfaceFormat[8]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_DCI_P3_NONLINEAR_EXT
                SurfaceFormat[9]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_DCI_P3_NONLINEAR_EXT
                SurfaceFormat[10]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_DCI_P3_NONLINEAR_EXT
                SurfaceFormat[11]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_DCI_P3_NONLINEAR_EXT
                SurfaceFormat[12]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_BT709_NONLINEAR_EXT
                SurfaceFormat[13]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_BT709_NONLINEAR_EXT
                SurfaceFormat[14]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_BT709_NONLINEAR_EXT
                SurfaceFormat[15]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_BT709_NONLINEAR_EXT
                SurfaceFormat[16]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_ADOBERGB_NONLINEAR_EXT
                SurfaceFormat[17]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_ADOBERGB_NONLINEAR_EXT
                SurfaceFormat[18]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_ADOBERGB_NONLINEAR_EXT
                SurfaceFormat[19]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_ADOBERGB_NONLINEAR_EXT
                SurfaceFormat[20]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_PASS_THROUGH_EXT
                SurfaceFormat[21]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_PASS_THROUGH_EXT
                SurfaceFormat[22]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_PASS_THROUGH_EXT
                SurfaceFormat[23]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_PASS_THROUGH_EXT
                SurfaceFormat[24]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_EXTENDED_SRGB_LINEAR_EXT
                SurfaceFormat[25]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_EXTENDED_SRGB_LINEAR_EXT
                SurfaceFormat[26]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_EXTENDED_SRGB_LINEAR_EXT
                SurfaceFormat[27]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_EXTENDED_SRGB_LINEAR_EXT
                SurfaceFormat[28]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT
                SurfaceFormat[29]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT
                SurfaceFormat[30]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT
                SurfaceFormat[31]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT
                SurfaceFormat[32]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_DISPLAY_P3_LINEAR_EXT
                SurfaceFormat[33]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_DISPLAY_P3_LINEAR_EXT
                SurfaceFormat[34]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_DISPLAY_P3_LINEAR_EXT
                SurfaceFormat[35]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_DISPLAY_P3_LINEAR_EXT
                SurfaceFormat[36]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_BT2020_LINEAR_EXT
                SurfaceFormat[37]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_BT2020_LINEAR_EXT
                SurfaceFormat[38]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_BT2020_LINEAR_EXT
                SurfaceFormat[39]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_BT2020_LINEAR_EXT
                SurfaceFormat[40]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_HDR10_ST2084_EXT
                SurfaceFormat[41]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_HDR10_ST2084_EXT
                SurfaceFormat[42]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_HDR10_ST2084_EXT
                SurfaceFormat[43]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_HDR10_ST2084_EXT
                SurfaceFormat[44]:
                        format = FORMAT_B8G8R8A8_UNORM
                        colorSpace = COLOR_SPACE_HDR10_HLG_EXT
                SurfaceFormat[45]:
                        format = FORMAT_B8G8R8A8_SRGB
                        colorSpace = COLOR_SPACE_HDR10_HLG_EXT
                SurfaceFormat[46]:
                        format = FORMAT_R16G16B16A16_SFLOAT
                        colorSpace = COLOR_SPACE_HDR10_HLG_EXT
                SurfaceFormat[47]:
                        format = FORMAT_A2R10G10B10_UNORM_PACK32
                        colorSpace = COLOR_SPACE_HDR10_HLG_EXT
        Present Modes: count = 2
                PRESENT_MODE_FIFO_KHR
                PRESENT_MODE_IMMEDIATE_KHR
        VkSurfaceCapabilitiesKHR:
        -------------------------
                minImageCount       = 2
                maxImageCount       = 3
                currentExtent:
                        width  = 256
                        height = 256
                minImageExtent:
                        width  = 256
                        height = 256
                maxImageExtent:
                        width  = 256
                        height = 256
                maxImageArrayLayers = 1
                supportedTransforms:
                        SURFACE_TRANSFORM_IDENTITY_BIT_KHR
                currentTransform:
                        SURFACE_TRANSFORM_IDENTITY_BIT_KHR
                supportedCompositeAlpha:
                        COMPOSITE_ALPHA_OPAQUE_BIT_KHR
                        COMPOSITE_ALPHA_POST_MULTIPLIED_BIT_KHR
                        COMPOSITE_ALPHA_INHERIT_BIT_KHR
                supportedUsageFlags:
                        IMAGE_USAGE_TRANSFER_SRC_BIT
                        IMAGE_USAGE_TRANSFER_DST_BIT
                        IMAGE_USAGE_SAMPLED_BIT
                        IMAGE_USAGE_STORAGE_BIT
                        IMAGE_USAGE_COLOR_ATTACHMENT_BIT
        VkSurfaceCapabilities2EXT:
        --------------------------
                supportedSurfaceCounters:
                        None


Groups:
=======
        Device Group Properties (Group 0):
        ----------------------------------
                physicalDeviceCount: count = 1
                        Intel(R) Iris(TM) Plus Graphics 650 (ID: 0)
                subsetAllocation = 0

        Device Group Present Capabilities (Group 0):
                Intel(R) Iris(TM) Plus Graphics 650 (ID: 0)
                Can present images from the following devices:
                        Intel(R) Iris(TM) Plus Graphics 650 (ID: 0)
                Present modes:
                        DEVICE_GROUP_PRESENT_MODE_LOCAL_BIT_KHR


Device Properties and Extensions:
=================================
GPU0:
VkPhysicalDeviceProperties:
---------------------------
        apiVersion     = 4194430 (1.0.126)
        driverVersion  = 10038 (0x2736)
        vendorID       = 0x8086
        deviceID       = 0x5927
        deviceType     = PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU
        deviceName     = Intel(R) Iris(TM) Plus Graphics 650

VkPhysicalDeviceLimits:
-----------------------
        maxImageDimension1D                             = 16384
        maxImageDimension2D                             = 16384
        maxImageDimension3D                             = 2048
        maxImageDimensionCube                           = 16384
        maxImageArrayLayers                             = 2048
        maxTexelBufferElements                          = 67108864
        maxUniformBufferRange                           = 65536
        maxStorageBufferRange                           = 2147483648
        maxPushConstantsSize                            = 4096
        maxMemoryAllocationCount                        = 1073741823
        maxSamplerAllocationCount                       = 1073741823
        bufferImageGranularity                          = 0x00000100
        sparseAddressSpaceSize                          = 0x00000000
        maxBoundDescriptorSets                          = 1073741823
        maxPerStageDescriptorSamplers                   = 16
        maxPerStageDescriptorUniformBuffers             = 31
        maxPerStageDescriptorStorageBuffers             = 31
        maxPerStageDescriptorSampledImages              = 128
        maxPerStageDescriptorStorageImages              = 128
        maxPerStageDescriptorInputAttachments           = 128
        maxPerStageResources                            = 159
        maxDescriptorSetSamplers                        = 64
        maxDescriptorSetUniformBuffers                  = 124
        maxDescriptorSetUniformBuffersDynamic           = 124
        maxDescriptorSetStorageBuffers                  = 124
        maxDescriptorSetStorageBuffersDynamic           = 124
        maxDescriptorSetSampledImages                   = 512
        maxDescriptorSetStorageImages                   = 512
        maxDescriptorSetInputAttachments                = 512
        maxVertexInputAttributes                        = 31
        maxVertexInputBindings                          = 31
        maxVertexInputAttributeOffset                   = 4096
        maxVertexInputBindingStride                     = 4095
        maxVertexOutputComponents                       = 128
        maxTessellationGenerationLevel                  = 64
        maxTessellationPatchSize                        = 32
        maxTessellationControlPerVertexInputComponents  = 128
        maxTessellationControlPerVertexOutputComponents = 128
        maxTessellationControlPerPatchOutputComponents  = 120
        maxTessellationControlTotalOutputComponents     = 4216
        maxTessellationEvaluationInputComponents        = 128
        maxTessellationEvaluationOutputComponents       = 128
        maxGeometryShaderInvocations                    = 0
        maxGeometryInputComponents                      = 0
        maxGeometryOutputComponents                     = 0
        maxGeometryOutputVertices                       = 0
        maxGeometryTotalOutputComponents                = 0
        maxFragmentInputComponents                      = 128
        maxFragmentOutputAttachments                    = 8
        maxFragmentDualSrcAttachments                   = 1
        maxFragmentCombinedOutputResources              = 159
        maxComputeSharedMemorySize                      = 65536
        maxComputeWorkGroupCount: count = 3
                1073741823
                1073741823
                1073741823
        maxComputeWorkGroupInvocations                  = 1024
        maxComputeWorkGroupSize: count = 3
                1024
                1024
                1024
        subPixelPrecisionBits                           = 4
        subTexelPrecisionBits                           = 4
        mipmapPrecisionBits                             = 4
        maxDrawIndexedIndexValue                        = 4294967295
        maxDrawIndirectCount                            = 1073741823
        maxSamplerLodBias                               = 2
        maxSamplerAnisotropy                            = 16
        maxViewports                                    = 16
        maxViewportDimensions: count = 2
                16384
                16384
        viewportBoundsRange: count = 2
                -32768
                32767
        viewportSubPixelBits                            = 0
        minMemoryMapAlignment                           = 256
        minTexelBufferOffsetAlignment                   = 0x00000010
        minUniformBufferOffsetAlignment                 = 0x00000100
        minStorageBufferOffsetAlignment                 = 0x00000004
        minTexelOffset                                  = -8
        maxTexelOffset                                  = 7
        minTexelGatherOffset                            = -8
        maxTexelGatherOffset                            = 7
        minInterpolationOffset                          = -0.5
        maxInterpolationOffset                          = 0.5
        subPixelInterpolationOffsetBits                 = 4
        maxFramebufferWidth                             = 16384
        maxFramebufferHeight                            = 16384
        maxFramebufferLayers                            = 256
        framebufferColorSampleCounts:
                SAMPLE_COUNT_1_BIT
                SAMPLE_COUNT_2_BIT
                SAMPLE_COUNT_4_BIT
                SAMPLE_COUNT_8_BIT
        framebufferDepthSampleCounts:
                SAMPLE_COUNT_1_BIT
                SAMPLE_COUNT_2_BIT
                SAMPLE_COUNT_4_BIT
                SAMPLE_COUNT_8_BIT
        framebufferStencilSampleCounts:
                SAMPLE_COUNT_1_BIT
                SAMPLE_COUNT_2_BIT
                SAMPLE_COUNT_4_BIT
                SAMPLE_COUNT_8_BIT
        framebufferNoAttachmentsSampleCounts:
                SAMPLE_COUNT_1_BIT
                SAMPLE_COUNT_2_BIT
                SAMPLE_COUNT_4_BIT
                SAMPLE_COUNT_8_BIT
        maxColorAttachments                             = 8
        sampledImageColorSampleCounts:
                SAMPLE_COUNT_1_BIT
                SAMPLE_COUNT_2_BIT
                SAMPLE_COUNT_4_BIT
                SAMPLE_COUNT_8_BIT
        sampledImageIntegerSampleCounts:
                SAMPLE_COUNT_1_BIT
                SAMPLE_COUNT_2_BIT
                SAMPLE_COUNT_4_BIT
                SAMPLE_COUNT_8_BIT
        sampledImageDepthSampleCounts:
                SAMPLE_COUNT_1_BIT
                SAMPLE_COUNT_2_BIT
                SAMPLE_COUNT_4_BIT
                SAMPLE_COUNT_8_BIT
        sampledImageStencilSampleCounts:
                SAMPLE_COUNT_1_BIT
                SAMPLE_COUNT_2_BIT
                SAMPLE_COUNT_4_BIT
                SAMPLE_COUNT_8_BIT
        storageImageSampleCounts:
                SAMPLE_COUNT_1_BIT
        maxSampleMaskWords                              = 1
        timestampComputeAndGraphics                     = true
        timestampPeriod                                 = 1
        maxClipDistances                                = 1073741823
        maxCullDistances                                = 0
        maxCombinedClipAndCullDistances                 = 1073741823
        discreteQueuePriorities                         = 2
        pointSizeRange: count = 2
                1
                511
        lineWidthRange: count = 2
                1
                1
        pointSizeGranularity                            = 1
        lineWidthGranularity                            = 1
        strictLines                                     = false
        standardSampleLocations                         = true
        optimalBufferCopyOffsetAlignment                = 0x00000100
        optimalBufferCopyRowPitchAlignment              = 0x00000001
        nonCoherentAtomSize                             = 0x00000100

VkPhysicalDeviceSparseProperties:
---------------------------------
        residencyStandard2DBlockShape            = false
        residencyStandard2DMultisampleBlockShape = false
        residencyStandard3DBlockShape            = false
        residencyAlignedMipSize                  = false
        residencyNonResidentStrict               = false

VkPhysicalDeviceIDProperties:
-----------------------------
        deviceUUID      =  0000-00-00-00-000000
        driverUUID      =  0000-00-00-00-000000
        deviceNodeMask  = 0
        deviceLUIDValid = false

VkPhysicalDeviceMaintenance3Properties:
---------------------------------------
        maxPerSetDescriptors    = 700
        maxMemoryAllocationSize = 0x80000000

VkPhysicalDeviceMultiviewProperties:
------------------------------------
        maxMultiviewViewCount     = 0
        maxMultiviewInstanceIndex = 0

VkPhysicalDevicePointClippingProperties:
----------------------------------------
        pointClippingBehavior = POINT_CLIPPING_BEHAVIOR_ALL_CLIP_PLANES

VkPhysicalDeviceProtectedMemoryProperties:
------------------------------------------
        protectedNoFault = false

VkPhysicalDevicePushDescriptorPropertiesKHR:
--------------------------------------------
        maxPushDescriptors = 159

VkPhysicalDeviceSubgroupProperties:
-----------------------------------
        subgroupSize              = 0
        supportedStages:
                None
        supportedOperations:
                None
        quadOperationsInAllStages = false

VkPhysicalDeviceTexelBufferAlignmentPropertiesEXT:
--------------------------------------------------
        storageTexelBufferOffsetAlignmentBytes       = 0x00000010
        storageTexelBufferOffsetSingleTexelAlignment = true
        uniformTexelBufferOffsetAlignmentBytes       = 0x00000010
        uniformTexelBufferOffsetSingleTexelAlignment = true

VkPhysicalDeviceVertexAttributeDivisorPropertiesEXT:
----------------------------------------------------
        maxVertexAttribDivisor = 1073741823


Device Extensions: count = 37
        VK_AMD_gpu_shader_half_float          : extension revision 2
        VK_AMD_negative_viewport_height       : extension revision 1
        VK_AMD_shader_trinary_minmax          : extension revision 1
        VK_EXTX_portability_subset            : extension revision 1
        VK_EXT_debug_marker                   : extension revision 4
        VK_EXT_fragment_shader_interlock      : extension revision 1
        VK_EXT_hdr_metadata                   : extension revision 2
        VK_EXT_host_query_reset               : extension revision 1
        VK_EXT_memory_budget                  : extension revision 1
        VK_EXT_scalar_block_layout            : extension revision 1
        VK_EXT_shader_stencil_export          : extension revision 1
        VK_EXT_shader_viewport_index_layer    : extension revision 1
        VK_EXT_texel_buffer_alignment         : extension revision 1
        VK_EXT_vertex_attribute_divisor       : extension revision 3
        VK_INTEL_shader_integer_functions2    : extension revision 1
        VK_KHR_16bit_storage                  : extension revision 1
        VK_KHR_8bit_storage                   : extension revision 1
        VK_KHR_bind_memory2                   : extension revision 1
        VK_KHR_dedicated_allocation           : extension revision 3
        VK_KHR_descriptor_update_template     : extension revision 1
        VK_KHR_device_group                   : extension revision 4
        VK_KHR_get_memory_requirements2       : extension revision 1
        VK_KHR_image_format_list              : extension revision 1
        VK_KHR_maintenance1                   : extension revision 2
        VK_KHR_maintenance2                   : extension revision 1
        VK_KHR_maintenance3                   : extension revision 1
        VK_KHR_push_descriptor                : extension revision 2
        VK_KHR_relaxed_block_layout           : extension revision 1
        VK_KHR_sampler_mirror_clamp_to_edge   : extension revision 3
        VK_KHR_shader_draw_parameters         : extension revision 1
        VK_KHR_shader_float16_int8            : extension revision 1
        VK_KHR_storage_buffer_storage_class   : extension revision 1
        VK_KHR_swapchain                      : extension revision 70
        VK_KHR_swapchain_mutable_format       : extension revision 1
        VK_KHR_uniform_buffer_standard_layout : extension revision 1
        VK_KHR_variable_pointers              : extension revision 1
        VK_NV_glsl_shader                     : extension revision 1

VkQueueFamilyProperties[0]:
==========================
        minImageTransferGranularity = (1, 1, 1)
        queueCount                  = 1
        queueFlags                  = QUEUE_GRAPHICS | QUEUE_COMPUTE | QUEUE_TRANSFER
        timestampValidBits          = 64
        present support = false

VkQueueFamilyProperties[1]:
==========================
        minImageTransferGranularity = (1, 1, 1)
        queueCount                  = 1
        queueFlags                  = QUEUE_GRAPHICS | QUEUE_COMPUTE | QUEUE_TRANSFER
        timestampValidBits          = 64
        present support = false

VkQueueFamilyProperties[2]:
==========================
        minImageTransferGranularity = (1, 1, 1)
        queueCount                  = 1
        queueFlags                  = QUEUE_GRAPHICS | QUEUE_COMPUTE | QUEUE_TRANSFER
        timestampValidBits          = 64
        present support = false

VkQueueFamilyProperties[3]:
==========================
        minImageTransferGranularity = (1, 1, 1)
        queueCount                  = 1
        queueFlags                  = QUEUE_GRAPHICS | QUEUE_COMPUTE | QUEUE_TRANSFER
        timestampValidBits          = 64
        present support = false

VkPhysicalDeviceMemoryProperties:
=================================
memoryHeaps: count = 1
        memoryHeaps[0]:
                size   = 8589934592 (0x200000000) (8.00 GiB)
                budget = 1610612736
                usage  = 524288
                flags:
                        MEMORY_HEAP_DEVICE_LOCAL_BIT
memoryTypes: count = 3
        memoryTypes[0]:
                heapIndex     = 0
                propertyFlags = 0x0001:
                        MEMORY_PROPERTY_DEVICE_LOCAL_BIT
                usable for:
                        IMAGE_TILING_OPTIMAL: color images, FORMAT_D16_UNORM, FORMAT_D32_SFLOAT, FORMAT_S8_UINT, FORMAT_D32_SFLOAT_S8_UINT
                        IMAGE_TILING_LINEAR: None
        memoryTypes[1]:
                heapIndex     = 0
                propertyFlags = 0x000b:
                        MEMORY_PROPERTY_DEVICE_LOCAL_BIT
                        MEMORY_PROPERTY_HOST_VISIBLE_BIT
                        MEMORY_PROPERTY_HOST_CACHED_BIT
                usable for:
                        IMAGE_TILING_OPTIMAL: color images
                        IMAGE_TILING_LINEAR: None
        memoryTypes[2]:
                heapIndex     = 0
                propertyFlags = 0x000f:
                        MEMORY_PROPERTY_DEVICE_LOCAL_BIT
                        MEMORY_PROPERTY_HOST_VISIBLE_BIT
                        MEMORY_PROPERTY_HOST_COHERENT_BIT
                        MEMORY_PROPERTY_HOST_CACHED_BIT
                usable for:
                        IMAGE_TILING_OPTIMAL: None
                        IMAGE_TILING_LINEAR: None

VkPhysicalDeviceFeatures:
=========================
        robustBufferAccess                      = true
        fullDrawIndexUint32                     = true
        imageCubeArray                          = true
        independentBlend                        = true
        geometryShader                          = false
        tessellationShader                      = true
        sampleRateShading                       = true
        dualSrcBlend                            = true
        logicOp                                 = false
        multiDrawIndirect                       = true
        drawIndirectFirstInstance               = true
        depthClamp                              = true
        depthBiasClamp                          = true
        fillModeNonSolid                        = true
        depthBounds                             = false
        wideLines                               = false
        largePoints                             = true
        alphaToOne                              = true
        multiViewport                           = true
        samplerAnisotropy                       = true
        textureCompressionETC2                  = false
        textureCompressionASTC_LDR              = false
        textureCompressionBC                    = true
        occlusionQueryPrecise                   = true
        pipelineStatisticsQuery                 = false
        vertexPipelineStoresAndAtomics          = true
        fragmentStoresAndAtomics                = true
        shaderTessellationAndGeometryPointSize  = true
        shaderImageGatherExtended               = true
        shaderStorageImageExtendedFormats       = true
        shaderStorageImageMultisample           = false
        shaderStorageImageReadWithoutFormat     = true
        shaderStorageImageWriteWithoutFormat    = true
        shaderUniformBufferArrayDynamicIndexing = true
        shaderSampledImageArrayDynamicIndexing  = true
        shaderStorageBufferArrayDynamicIndexing = true
        shaderStorageImageArrayDynamicIndexing  = true
        shaderClipDistance                      = true
        shaderCullDistance                      = false
        shaderFloat64                           = false
        shaderInt64                             = false
        shaderInt16                             = true
        shaderResourceResidency                 = false
        shaderResourceMinLod                    = true
        sparseBinding                           = false
        sparseResidencyBuffer                   = false
        sparseResidencyImage2D                  = false
        sparseResidencyImage3D                  = false
        sparseResidency2Samples                 = false
        sparseResidency4Samples                 = false
        sparseResidency8Samples                 = false
        sparseResidency16Samples                = false
        sparseResidencyAliased                  = false
        variableMultisampleRate                 = true
        inheritedQueries                        = true

VkPhysicalDevice16BitStorageFeatures:
-------------------------------------
        storageBuffer16BitAccess           = true
        uniformAndStorageBuffer16BitAccess = true
        storagePushConstant16              = true
        storageInputOutput16               = true

VkPhysicalDevice8BitStorageFeaturesKHR:
---------------------------------------
        storageBuffer8BitAccess           = true
        uniformAndStorageBuffer8BitAccess = true
        storagePushConstant8              = true

VkPhysicalDeviceFragmentShaderInterlockFeaturesEXT:
---------------------------------------------------
        fragmentShaderSampleInterlock      = true
        fragmentShaderPixelInterlock       = true
        fragmentShaderShadingRateInterlock = false

VkPhysicalDeviceHostQueryResetFeaturesEXT:
------------------------------------------
        hostQueryReset = true

VkPhysicalDeviceMultiviewFeatures:
----------------------------------
        multiview                   = false
        multiviewGeometryShader     = false
        multiviewTessellationShader = false

VkPhysicalDeviceProtectedMemoryFeatures:
----------------------------------------
        protectedMemory = false

VkPhysicalDeviceSamplerYcbcrConversionFeatures:
-----------------------------------------------
        samplerYcbcrConversion = false

VkPhysicalDeviceScalarBlockLayoutFeaturesEXT:
---------------------------------------------
        scalarBlockLayout = true

VkPhysicalDeviceShaderDrawParametersFeatures:
---------------------------------------------
        shaderDrawParameters = false

VkPhysicalDeviceShaderFloat16Int8FeaturesKHR:
---------------------------------------------
        shaderFloat16 = true
        shaderInt8    = true

VkPhysicalDeviceTexelBufferAlignmentFeaturesEXT:
------------------------------------------------
        texelBufferAlignment = true

VkPhysicalDeviceUniformBufferStandardLayoutFeaturesKHR:
-------------------------------------------------------
        uniformBufferStandardLayout = true

VkPhysicalDeviceVariablePointersFeatures:
-----------------------------------------
        variablePointersStorageBuffer = true
        variablePointers              = true

VkPhysicalDeviceVertexAttributeDivisorFeaturesEXT:
--------------------------------------------------
        vertexAttributeInstanceRateDivisor     = true
        vertexAttributeInstanceRateZeroDivisor = true
```

**Comment by codeidoscope** [comment]
[log.log](https://github.com/nannou-org/nannou/files/3827461/log.log)

**Comment by freesig** [comment]
If someone could try running these examples on a system where nannou doesn't work and let me know if that is working that would be super helpful https://github.com/vulkano-rs/vulkano-examples

**Comment by teh-f00l** [comment]
This is what I get when running `cargo run --bin triangle` in the vulkano-examples repo:

<img width="1022" alt="VulkanTriangle" src="https://user-images.githubusercontent.com/36428156/68532670-a167e980-0320-11ea-80c8-4ca4b16745ce.png">

**Comment by codeidoscope** [comment]
I got the following:
```[Rest of the things that were downloaded and compiled with no issues]
   Compiling vulkano-shaders v0.13.0
   Compiling examples v0.1.0 (/Users/marion/Desktop/Engineering/rustfest/vulkano-examples)
    Finished dev [unoptimized + debuginfo] target(s) in 3m 18s
     Running `target/debug/triangle`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: LoadingError(LibraryLoadFailure("dlopen(libvulkan.1.dylib, 1): image not found"))', src/libcore/result.rs:1165:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.```

**Comment by freesig** [comment]
@codeidoscope which OS are you on?

**Comment by teh-f00l** [comment]
Just confirmed animation works too in `Vulkano-Examples` with   `cargo run --bin teapot`

**Comment by codeidoscope** [comment]
> @codeidoscope which OS are you on?

MacOS Catalina Version 10.15.1 (19B88)
https://github.com/nannou-org/nannou/issues/423#issuecomment-552113538

**Comment by freesig** [comment]
Did you get a pop asking to allow a dynamic library. I think the only options are `move to bin` and `cancel`?
You have go cancel then go to system preferences / security and allow it then re-run.
Sorry Catalina is really sandboxed now :(

**Comment by codeidoscope** [comment]
@freesig I had a bunch of them when I installed Vulkano yesterday in the workshop (they were all allowed), but I haven't had a new one appear when I ran it again. The error message appears for all the Vulkano examples, if that helps?

**Comment by freesig** [comment]
Yeh you might need to remove the ~/.vulkano_sdk directory and install it again manually. I'd say a library or binary got deleted from there but it's hard to know which one.

**Comment by nestarz** [comment]
I have the same issue using the same setup (Catalina, Intel Plus Graphics 650). Is there a workaround to make thee nannou (examples included) working ?

**Comment by mrutyunjayan** [comment]
has a fix for this been found? I tried running the example today on Catalina and my animations looked weird as well

**Comment by mitchmindtree** [comment]
Hi folks, sorry for the delay in addressing this! #452 should fix this once it lands - I'm currently just attempting to fix some stuttery loop timing issues, then need to update the examples/docs/guide and then it should be ready for wider testing and landing.

**Comment by mitchmindtree** [comment]
OK #452 should be ready for testing! I've tested it on Mojave and it's working smoothly. I haven't tested it on Catalina but I suspect any graphical glitches there should also be fixed.

If you're interested in testing #452 but are unsure how to do so, [this might help](https://github.com/TeamPorcupine/ProjectPorcupine/wiki/How-to-Test-a-Pull-Request).

**Comment by mitchmindtree** [comment]
This should now be closed by #452.
